xwakes/wit/utilities.py

```diff
@@ -698,35 +376,140 @@ def create_taper_RW_approx_element(
+    the source particle. Also
+    referred to as 'a' and 'b'. Must be specified for valid initialization
+    :param name: An optional user-specified name of the component
+    :param f_rois: a list of frequency regions of interest
+    :param t_rois: a list of time regions of interest
     """
-    Creates a component in which the impedance function is evaluated directly only on few points and it is interpolated
-    everywhere else. This helps when the impedance function is very slow to evaluate.
-    :param interpolation_frequencies: the frequencies where the function is evaluated for the interpolation
-    :param impedance: A callable function representing the impedance function of the Component. Can be undefined if
+    return ComponentInterpolated(impedance_input=impedance, wake_input=wake,
+                                 plane=plane, source_exponents=source_exponents,
+                                 test_exponents=test_exponents, name=name,
+                                 interpolation_frequencies=interpolation_frequencies,
+                                 interpolation_times=interpolation_times,
+                                 f_rois=f_rois, t_rois=t_rois)
+
+
+def create_component_from_arrays(interpolation_frequencies: ArrayLike = None,
+                                 impedance_samples: ArrayLike = None,
+                                 interpolation_times: ArrayLike = None,
+                                 wake_samples: ArrayLike = None,
+                                 plane: str = None,
+                                 source_exponents: Tuple[int, int] = None,
+                                 test_exponents: Tuple[int, int] = None,
+                                 name: str = "Interpolated Component",
+                                 f_rois: Optional[List[Tuple[float, float]]] = None,
+                                 t_rois: Optional[List[Tuple[float, float]]] = None):
+    """
+    Creates a component from impedance and/or wake functions defined discretely
+    through arrays
+    :param interpolation_frequencies: the frequencies where the impedance
+    function is evaluated for the interpolation
+    :param impedance: A callable function representing the impedance function of
+    the Component. Can be undefined if
     the wake function is defined.
-    :param wake: A callable function representing the wake function of the Component. Can be undefined if
-    the impedance function is defined.
-    :param plane: The plane of the Component, either 'x', 'y' or 'z'. Must be specified for valid initialization
-    :param source_exponents: The exponents in the x and y planes experienced by the source particle. Also
+    :param interpolation_times: the times where the wake function is evaluated
+    for the interpolation
+    :param wake: A callable function representing the wake function of the
+    Component. Can be undefined if the impedance function is defined.
```

Comment:
`impedance` and `wake` are not in the inputs, and the `samples` inputs are missing from the docstring (probably some remnant of a copy-paste)